### PR TITLE
drivers: dmic: clear SOFT_RESET bit to ACE as well

### LINKED ISSUE #6723: [BUG] I/O error with DMIC capture pause/release on MTL
**Describe the bug**
CI observed an Input/output error when check-pause-resume-capture-100  or check-signal-stop-start.sh on MTL platforms.

**To Reproduce**
```
TPLG=/lib/firmware/intel/sof-ace-tplg/sof-mtl-rt711-4ch.tplg MODEL=MTLP_RVP_SDW ~/sof-test/test-case/check-pause-resume.sh -c 100 -m capture
```

**Reproduction Rate**
100%

**Environment**
1) Branch name and commit hash of the 2 repositories: sof (firmware/topology) and linux (kernel driver).
    * Kernel: https://github.com/thesofproject/linux/commit/321028577de9
    * SOF: https://github.com/thesofproject/sof/commit/4a9a99aeb24a
2) Name of the topology file
    * Topology: sof-ace-tplg/sof-mtl-rt711-4ch.tplg, sof-ace-tplg/sof-hda-generic-4ch.tplg
3) Name of the platform(s) on which the bug is observed.
    * Platform: MTLP_RVP_SDW, MTLP_RVP_HDA

**Screenshots or console output**

**console output**
```
=== PAUSE ===                                                            
(100/100) Wait for 121 ms before pause

(100/100) Wait for 181 ms before resume

declare -- cmd="journalctl_cmd --since=@1670281836"
2022-12-05 23:11:08 UTC [REMOTE_INFO] Entering expect script with:
      arecord   -D hw:0,10 -r 48000 -c 4 -f S32_LE -vv -i /dev/null -q
spawn arecord -D hw:0,10 -r 48000 -c 4 -f S32_LE -vv -i /dev/null -q
Hardware PCM card 0 'sof-soundwire' device 10 subdevice 0
Its setup is:
  stream       : CAPTURE
  access       : RW_INTERLEAVED
  format       : S32_LE
  subformat    : STD
  channels     : 4
  rate         : 48000
  exact rate   : 48000 (48000/1)
  msbits       : 32
  buffer_size  : 4096
  period_size  : 1024
  period_time  : 21333
  tstamp_mode  : NONE
  tstamp_type  : MONOTONIC
  period_step  : 1
  avail_min    : 1024
  period_event : 0
  start_threshold  : 1
  stop_threshold   : 4096
  silence_threshold: 0
  silence_size : 0
  boundary     : 4611686018427387904
  appl_ptr     : 0
  hw_ptr       : 0

#+                                                 | 00%
(1/100) Wait for 142 ms before pause
 
#+                                                 | 00%
=== PAUSE ===                                                            
(1/100) Wait for 107 ms before pause

(1/100) Wait for 104 ms before resume
                                                                           
 
#+                                                 | 00%
=== PAUSE ===                                                            
(2/100) Wait for 189 ms before pause

(2/100) Wait for 181 ms before resume
                                                                           
 arecord: pcm_read:2221: read error: Input/output error
2022-12-05 23:11:10 UTC [REMOTE_ERROR] Starting func_exit_handler(), exit status=1, FUNCNAME stack:
2022-12-05 23:11:10 UTC [REMOTE_ERROR]  main()  @  /home/ubuntu/sof-test/test-case/check-pause-resume.sh
```

[full console_log.txt](https://github.com/thesofproject/sof/files/10161836/console_log.txt)
[mtrace.txt](https://github.com/thesofproject/sof/files/10161834/mtrace.txt)
[dmesg.txt](https://github.com/thesofproject/sof/files/10161835/dmesg.txt)

### PR DESCRIPTION
**Pull request by fredoh9**
Clearing sof-reset bit applies to ACE as well. And when releasing a paused stream, unmuting mic is necessary. Without unmuting there will be no input data after resume.

fixes: https://github.com/thesofproject/sof/issues/6723

### CODE REVIEW

**Review by ranj063 — DISMISSED** [review]
Thanks @fredoh9 !

**Comment by teburd** [comment]
nice find @ranj063 @fredoh9 !

**Comment by dnikodem** [comment]
> @dnikodem can you run your internal tests with this? I remember this was like this because mtl was not very happy without it...

Sure. I have asked our validation to help with this. I will let you know when the results are ready. Thanks

File: drivers/dai/intel/dmic/dmic.c [hunk 1]
```diff
@@ -607,18 +607,16 @@ static void dai_dmic_start(struct dai_intel_dmic *dmic)
...
 	}
 
-#ifndef CONFIG_SOC_SERIES_INTEL_ACE
 	/* Clear soft reset for all/used PDM controllers. This should
 	 * start capture in sync.
 	 */
 	for (i = 0; i < CONFIG_DAI_DMIC_HW_CONTROLLERS; i++) {
 		dai_dmic_update_bits(dmic, base[i] + CIC_CONTROL,
-				     CIC_CONTROL_SOFT_RESET_BIT, 0);
+				     CIC_CONTROL_SOFT_RESET_BIT|CIC_CONTROL_MIC_MUTE_BIT, 0);
```
Comment (kv2019i) [inline drivers/dai/intel/dmic/dmic.c:615]: I wonder if it would be better to have two separate commits, one to apply this code on ACE as well, and second commit to add the MUTE bit. 

Minor style note: empty space around bit operations is preferred, i.e. "CIC_CONTROL_SOFT_RESET_BIT | CIC_CONTROL_MIC_MUTE_BIT"

**Review by kv2019i — COMMENTED** [review]
Good catch @fredoh9 ! Some minor notes inline.

**Comment by juimonen** [comment]
> @juimonen @fredoh9 @dnikodem using FW from [thesofproject/sof#7332](https://github.com/thesofproject/sof/pull/7332), I checked the DMIC on MTL and found no regression. Everything seems fine

@wszypelt thanks! then I'm ok with this PR.

File: drivers/dai/intel/dmic/dmic.c [hunk 2]
```diff
@@ -607,18 +607,16 @@ static void dai_dmic_start(struct dai_intel_dmic *dmic)
 		}
 	}
 
-#ifndef CONFIG_SOC_SERIES_INTEL_ACE
 	/* Clear soft reset for all/used PDM controllers. This should
 	 * start capture in sync.
 	 */
 	for (i = 0; i < CONFIG_DAI_DMIC_HW_CONTROLLERS; i++) {
 		dai_dmic_update_bits(dmic, base[i] + CIC_CONTROL,
-				     CIC_CONTROL_SOFT_RESET_BIT, 0);
```
[the remark this answers is not in the document]
Comment (plbossart) [reply drivers/dai/intel/dmic/dmic.c:616]: @singalsu can you clarify your comment? it's not clear to me if you are suggesting another solution and based on what?

If the blobs should be changed that's not a zephyr change but a topology plugin change to handled by @juimonen

File: drivers/dai/intel/dmic/dmic.c (hunk 1 above)
Comment (fredoh9) [reply drivers/dai/intel/dmic/dmic.c:615]: I'm still working on it. I have some cases are failing today, trying to make sure everything is covered by this patch

**Review by ranj063 — APPROVED** [review]

File: drivers/dai/intel/dmic/dmic.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (fredoh9) [reply drivers/dai/intel/dmic/dmic.c]: @singalsu I tried below diff instead of second commit, but it didn't work. I have only zero input. Can you show me what I missed?

```diff
--- a/drivers/dai/intel/dmic/dmic.c
+++ b/drivers/dai/intel/dmic/dmic.c
@@ -604,12 +604,20 @@ static void dai_dmic_start(struct dai_intel_dmic *dmic)
                switch (dmic->dai_config_params.dai_index) {
                case 0:
                        dai_dmic_update_bits(dmic, base[i] + FIR_CONTROL_A,
+#ifdef CONFIG_SOC_SERIES_INTEL_ACE
+                                            FIR_CONTROL_A_START_BIT | FIR_CONTROL_A_PERIODIC_START_EN_BIT,
+#else
                                             FIR_CONTROL_A_START_BIT,
+#endif
                                             FIR_CONTROL_A_START(fir_a));
                        break;
                case 1:
                        dai_dmic_update_bits(dmic, base[i] + FIR_CONTROL_B,
+#ifdef CONFIG_SOC_SERIES_INTEL_ACE
+                                            FIR_CONTROL_B_START_BIT | FIR_CONTROL_B_PERIODIC_START_EN_BIT,
+#else
                                             FIR_CONTROL_B_START_BIT,
+#endif
                                             FIR_CONTROL_B_START(fir_b));
                        break;
                }
```

**Review by singalsu — APPROVED** [review]

**Review by abonislawski — APPROVED** [review]

**Review by kv2019i — APPROVED** [review]
Excellent, thanks @fredoh9 ! We might want to revisit whether INF is really needed for start, but I think given the amount of issues with the dmic start flows, I think it makes sense to have it in the logs and this anyways is only once per stream start.

**Review by juimonen — APPROVED** [review]
